src/index.js

```diff
@@ -8,31 +9,50 @@ import registerConfigAsDependency from './lib/registerConfigAsDependency'
 import processTailwindFeatures from './processTailwindFeatures'
 import resolveConfig from './util/resolveConfig'
 
-const plugin = postcss.plugin('tailwind', config => {
-  const plugins = []
+function resolveConfigPath(filePath) {
+  if (_.isObject(filePath)) {
+    return undefined
+  }
+
+  if (!_.isUndefined(filePath)) {
+    return path.resolve(filePath)
+  }
+
+  try {
+    const defaultConfigPath = path.resolve('./tailwind.config.js')
+    fs.accessSync(defaultConfigPath)
+    return defaultConfigPath
+  } catch (err) {
+    return undefined
+  }
+}
 
-  if (!_.isUndefined(config) && !_.isObject(config)) {
-    plugins.push(registerConfigAsDependency(path.resolve(config)))
+const getConfigFunction = config => () => {
+  if (_.isUndefined(config) && !_.isObject(config)) {
+    return resolveConfig([require('../defaultConfig')()])
   }
 
-  const getConfig = () => {
-    if (_.isUndefined(config)) {
-      return resolveConfig([require('../defaultConfig')()])
-    }
+  if (!_.isObject(config)) {
+    delete require.cache[require.resolve(config)]
+  }
 
-    if (!_.isObject(config)) {
-      delete require.cache[require.resolve(path.resolve(config))]
-    }
+  return resolveConfig([
+    _.isObject(config) ? config : require(config),
+    require('../defaultConfig')(),
+  ])
+}
+
+const plugin = postcss.plugin('tailwind', config => {
+  const plugins = []
+  const resolvedConfigPath = resolveConfigPath(config)
 
-    return resolveConfig([
-      _.isObject(config) ? config : require(path.resolve(config)),
-      require('../defaultConfig')(),
-    ])
+  if (!_.isUndefined(resolvedConfigPath)) {
+    plugins.push(registerConfigAsDependency(resolvedConfigPath))
   }
 
   return postcss([
     ...plugins,
-    processTailwindFeatures(getConfig),
+    processTailwindFeatures(getConfigFunction(resolvedConfigPath ? resolvedConfigPath : config)),
```

Comment:
@adamwathan You can just do `resolvedConfigPath || config`, no?

Reply:
Nice, made that change 👍